langchain4j-vertex-ai/src/test/java/dev/langchain4j/model/vertexai/VertexAiMatchingEngineIT.java

```diff
@@ -0,0 +1,61 @@
+package dev.langchain4j.model.vertexai;
+
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.store.embedding.CosineSimilarity;
+import dev.langchain4j.store.embedding.EmbeddingStore;
+import dev.langchain4j.store.embedding.EmbeddingStoreIT;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+
+import java.util.List;
+
+@Disabled("To run this test, you must have provide your own endpoint, project and location")
+public class VertexAiMatchingEngineIT extends EmbeddingStoreIT {
+
+    private VertexAiMatchingEngine matchingEngine;
+    private EmbeddingModel embeddingModel;
+
+    @BeforeEach
+    void beforeEach() {
+        embeddingModel = VertexAiEmbeddingModel.builder()
+                .endpoint("us-central1-aiplatform.googleapis.com:443")
+                .project("[PROJECT]")
+                .location("us-central1")
+                .publisher("google")
+                .modelName("textembedding-gecko@001")
+                .maxRetries(3)
+                .build();
+
+        matchingEngine = VertexAiMatchingEngine
+                .builder()
+                .endpoint("us-central1-aiplatform.googleapis.com:443")
+                .bucketName("[BUCKET_NAME]")
+                .indexId("[INDEX_ID]")
+                .deployedIndexId("[INDEX_DEPLOYED_ID]")
+                .indexEndpointId("[INDEX_ENDPOINT_ID]")
+                .location("us-central1")
+                .project("[PROJECT]")
+                .build();
+
+
+        final List<String> indices = matchingEngine.allIndices();
+        indices.forEach(matchingEngine::deleteIndex);
+    }
+
+    @Override
+    protected EmbeddingStore<TextSegment> embeddingStore() {
+        return matchingEngine;
+    }
+
+    @Override
+    protected EmbeddingModel embeddingModel() {
+        return embeddingModel;
+    }
+
+    @Override
+    protected double getCosineSimilarity(Embedding embedding, Embedding referenceEmbedding) {
+        return CosineSimilarity.between(embedding, referenceEmbedding);
```

Comment:
In VertexAI is depends on the algorithm you are using when creating the index. The one use default in the unit tests is not recommended by Google. But happy to revert back and put a note in the unit tests.

Reply:
which algorithms are supported and wich is the default one? can't find it in the code...

Reply:
It is defined during index creation. Here is different algorithm that can be configured:

**SQUARED_L2_DISTANCE**	        

>  Euclidean (L2) Distance

**L1_DISTANCE**

> Manhattan (L1) Distance

**DOT_PRODUCT_DISTANCE**	

> Default value. Defined as a negative of the dot product.

**COSINE_DISTANCE**	

> Cosine Distance. We strongly suggest using DOT_PRODUCT_DISTANCE + UNIT_L2_NORM instead of the COSINE distance. Our algorithms have been more optimized for the DOT_PRODUCT distance, and when combined with UNIT_L2_NORM, it offers the same ranking and mathematical equivalence as the COSINE distance.

Reply:
I am using the default one for testing!

Reply:
Each List<Embedding> has an index file attached to it. This index file is used by the matching engine to index it. If you create a batch mode index, then matching engine scan all the files to add the embeddings to the index. In short, an index is created in 2 phases, first create an index in matching engine (that is where you define the algorithm, ....), then you need to create files that represent a list of embeddings. I implemented the 2 phase, will probably do the first one but this one can be done manually using the UI. 

For the relevance, yes I can do that and assume that the index is created with the right algorithm.

By the way I looked the implementation from the python's langchain and they are doing the same.